lib/chef/resource/aws_cache_cluster.rb

```diff
@@ -0,0 +1,31 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsCacheCluster < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::ElastiCache
+
+  attribute :cluster_name, kind_of: String, name_attribute: true
+  attribute :az_mode, kind_of: String
+  attribute :preferred_availability_zone, kind_of: String
+  attribute :preferred_availability_zones,
```

Comment:
<sigh> Because AWS supports both attributes, it makes sense for us to allow both.  The plural version is also only supported for Memcached.  Lets just keep both here and pass them through to AWS, then AWS can bubble up any errors.